lib/matplotlib/tests/test_backend_qt.py

```diff
@@ -114,6 +114,42 @@ def test_fig_close(backend):
     assert init_figs == Gcf.figs
 
 
+@pytest.mark.backend('Qt5Agg')
+def test_fig_close_ioff(qt_module):
+    # Turn off interactive mode
+    plt.ioff()
+
+    # Create a figure
+    fig = plt.figure()
+
+    # Callback to exit event loop
+    def QuitApp(event):
+        from matplotlib.backends.backend_qt5 import qApp
+        qApp.exit()
+
+    # Connect callback to drawing (requires no user interaction)
+    fig.canvas.mpl_connect('draw_event', QuitApp)
+
+    # Save original SIGINT handler
+    import signal
+    original_signal = signal.getsignal(signal.SIGINT)
+
+    # Use our own SIGINT handler to be 100% sure this is working
+    def CustomHandler(signum, frame):
+        pass
+
+    signal.signal(signal.SIGINT, CustomHandler)
+
+    # Show window- fires callback and exits
+    plt.show()
```

Comment:
Instead `show` it may be more reliable to start the qt event loop.

Reply:
Did this in the new commit